File: app/src/main/java/danb/speedrunbrowser/api/objects/SearchResultItem.java
package danb.speedrunbrowser.api.objects;

import android.widget.TextView;

import java.net.MalformedURLException;
import java.net.URL;

public interface SearchResultItem {
    String getName();
    String getTypeName();

    URL getIconUrl() throws MalformedURLException;

    void applyTextView(TextView tv);
}

File: app/src/main/java/danb/speedrunbrowser/api/objects/SearchResultItem.kt
package danb.speedrunbrowser.api.objects

import android.widget.TextView

import java.net.MalformedURLException
import java.net.URL

interface SearchResultItem {
    val resolvedName: String
    val type: String

    val iconUrl: URL?

    fun applyTextView(tv: TextView)
}
